internal/patches/patches.go

```diff
@@ -0,0 +1,53 @@
+package patches
+
+import (
+	"encoding/json"
+	"sort"
+	"strings"
+
+	"github.com/giantswarm/microerror"
+	jsonpatch "gomodules.xyz/jsonpatch/v2"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/giantswarm/azure-admission-controller/pkg/mutator"
+)
+
+func GeneratePatchesFrom(originalJSON []byte, current runtime.Object) ([]mutator.PatchOperation, error) {
+	currentJSON, err := json.Marshal(current)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	var patches []mutator.PatchOperation
+	{
+		var jsonPatches []jsonpatch.JsonPatchOperation
+		jsonPatches, err = jsonpatch.CreatePatch(originalJSON, currentJSON)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		patches = make([]mutator.PatchOperation, 0, len(jsonPatches))
+		for _, patch := range jsonPatches {
+			patches = append(patches, mutator.PatchOperation(patch))
+		}
+
+		sort.SliceStable(patches, func(i, j int) bool {
```

Comment:
Without sorting, the order of the patches varies from one test run to another, which causes the tests to sometimes fail.

Reply:
Good catch :+1: